woo-razorpay.php

```diff
@@ -1824,6 +1849,140 @@ public function update1ccOrderWC(& $order, $wcOrderId, $razorpayPaymentId)
+                        }
+
+                    }
+                    else if(is_plugin_active('pw-woocommerce-gift-cards/pw-gift-cards.php'))
+                    {
+                        $result = $wpdb->get_row( $wpdb->prepare( "SELECT * FROM `{$wpdb->pimwick_gift_card}` WHERE `number` = %s", $giftcard['code'] ) );
+                        if($result != null){
+                            $balance = $wpdb->get_var( $wpdb->prepare( "SELECT SUM(amount) FROM {$wpdb->pimwick_gift_card_activity} WHERE pimwick_gift_card_id = %d", $result->pimwick_gift_card_id ) );
+                            
+                            if($balance == null && $balance >= 0 && $usedAmt > $balance ){
+                                // initiate refund in case gift card faliure
+                                $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                            }else{
+                                //Deduct amount of gift card
+                               $this->debitGiftCards($orderId, $order, "order_id: $orderId checkout_update_order_meta", $usedAmt);
+                            }
+
+                        }else{
+                            $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                        }
+
+                    }else{
+                       $this->process_refund($orderId, $razorpayData['amount_paid'], $reason = '', $razorpayPaymentId);
+                    }
+                    
+                }else{
+
+                    $couponKey = $giftcard['code'];
+
+                    if (empty($couponKey) === false)
+                    {
+                        // Remove the same coupon, if already being added to order.
+                        $order->remove_coupon($couponKey);
+
+                        //TODO: Convert all razorpay amount in paise to rupees
+                        $discount_total = $giftcard['value']/100;
+
+                        //TODO: Verify source code implementation
+                        // Loop through products and apply the coupon discount
+                        foreach($order->get_items() as $order_item)
+                        {
+                            $total = $order_item->get_total();
+                            $order_item->set_subtotal($total);
+                            $order_item->set_total($total - $discount_total);
+                            $order_item->save();
+                        }
+                        // TODO: Test if individual use coupon fails by hardcoding here
+                        $isApplied = $order->apply_coupon($couponKey);
+                        $order->save();
+
+                        rzpLogInfo("Coupon details updated for orderId: $orderId");
+                    }
+                }
+                
+            }
+        }
+
+        public function debitGiftCards( $orderId, $order, $note, $usedAmt, $giftCardNo) {
```

Comment:
If this plugin only used in this class, make it protected.

Reply:
ok